x/genutil/src/gentx.rs

```diff
@@ -170,21 +170,12 @@ impl TxHandler for GentxTxHandler {
         let pub_key = match pub_key {
             Some(var) => PublicKey::from(var),
             None => {
-                #[derive(serde::Deserialize)]
-                struct NodeKeyJson {
-                    pub priv_key: gears::tendermint::crypto::PrivateKey,
-                }
-
-                let key: NodeKeyJson = serde_json::from_reader(std::fs::File::open(
-                    client_tx_context.home.join("config/node_key.json"),
-                )?)?;
-
-                match key.priv_key {
-                    gears::tendermint::crypto::PrivateKey::Ed25519(var) => PublicKey::Ed25519(
-                        Ed25519PubKey::try_from(var.verification_key().as_bytes().to_vec())?,
-                    ),
-                    _ => unreachable!(),
-                }
+                let file = std::fs::File::open(
+                    client_tx_context
+                        .home
+                        .join("config/priv_validator_key.json"), // TODO: delegate this to tendermint crate
+                )?;
+                tendermint::get_validator_pub_key(file).unwrap().into()
```

Comment:
Unwrap? Maybe return error, but place todo?

Reply:
Ooops forgot to fix that... fixed